raiden/tests/utils/mediation_fees.py

```diff
@@ -0,0 +1,104 @@
+from itertools import islice
+
+from raiden.exceptions import UndefinedMediationFee
+from raiden.transfer.channel import get_balance
+from raiden.transfer.mediated_transfer.mediation_fee import FeeScheduleState
+from raiden.transfer.state import NettingChannelState
+from raiden.utils.typing import (
+    Balance,
+    FeeAmount,
+    Iterable,
+    List,
+    Optional,
+    PaymentAmount,
+    PaymentWithFeeAmount,
+    Sequence,
+)
+
+
+def window(seq: Sequence, n: int = 2) -> Iterable[tuple]:
+    """Returns a sliding window (of width n) over data from the iterable
+    s -> (s0,s1,...s[n-1]), (s1,s2,...,sn), ...
+    See https://stackoverflow.com/a/6822773/114926
+    """
+    remaining_elements = iter(seq)
+    result = tuple(islice(remaining_elements, n))
+    if len(result) == n:
+        yield result
+    for elem in remaining_elements:
+        result = result[1:] + (elem,)
+        yield result
+
+
+def fee_sender(
+    fee_schedule: FeeScheduleState, balance: Balance, amount: PaymentWithFeeAmount
+) -> Optional[FeeAmount]:
+    """Returns the mediation fee for this channel when transferring the given amount"""
+    try:
+        return fee_schedule.fee_payer(amount, balance)
+    except UndefinedMediationFee:
+        return None
+
+
+def fee_receiver(
+    fee_schedule: FeeScheduleState, balance: Balance, amount: PaymentWithFeeAmount
+) -> Optional[FeeAmount]:
+    """Returns the mediation fee for this channel when receiving the given amount"""
+
+    def fee_in(imbalance_fee: FeeAmount) -> FeeAmount:
+        return FeeAmount(
+            round(
+                amount
+                - (
+                    (amount + fee_schedule.flat + imbalance_fee)
+                    / (1 - fee_schedule.proportional / 1e6)
+                )
+            )
+        )
+
+    imbalance_fee = fee_schedule.imbalance_fee(
+        amount=PaymentWithFeeAmount(amount - fee_in(imbalance_fee=FeeAmount(0))), balance=balance
+    )
+
+    return fee_in(imbalance_fee=imbalance_fee)
+
+
+def calculate_initial_amount_for_final_amount(
```

Comment:
name sounds a bit strange. Can't say a good name without it sounding too long:

`get_initial_payment_for_final_target_amount()` ?